Controller/CRUDController.php

```diff
@@ -119,13 +119,13 @@ public function listAction()
     /**
      * Execute a batch delete.
      *
-     * @param ProxyQueryInterface $query
+     * @param ProxyQueryInterface|null $query
      *
      * @return RedirectResponse
      *
      * @throws AccessDeniedException If access is not granted
      */
-    public function batchActionDelete(ProxyQueryInterface $query)
+    public function batchActionDelete(ProxyQueryInterface $query = null)
```

Comment:
This is a BC-break for extending classes

Reply:
So `$query = null;` line 468 should never be here.

Should we redirect to the list view instead of calling the finalAction ?

Reply:
I guess so, if there's nothing to do, let's not do anything, shall we?

Reply:
Have you seen this commit https://github.com/sonata-project/SonataAdminBundle/pull/4737/commits/cd3c4561260f3a877b509a4f75bf5bf5ab307bd5 ?